### PR TITLE
Support re-adding an archived item to My List

### PR DESCRIPTION
**Pull request by jacobsimeon**

### CODE REVIEW

File: PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift [hunk 1]
```diff
@@ -132,9 +132,8 @@ extension ArchivedItemsListViewModel {
     func overflowActions(for itemID: String) -> [ItemAction]? {
         return bareItem(with: itemID).flatMap { archivedItem in
             return [
-                .delete { [weak self] _ in
-                    self?.confirmDelete(item: archivedItem)
-                }
+                .delete { [weak self] _ in self?.confirmDelete(item: archivedItem) },
+                .reAdd { [weak self] _ in self?.reAdd(item: archivedItem) }
```
Comment (dskuza) [inline PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift:136]: Minor detail, but I'm wondering if the order of these should be flipped.

File: PocketKit/Sources/PocketKit/ItemAction.swift [hunk 2]
```diff
@@ -57,6 +57,16 @@ extension ItemAction {
         )
     }
 
+    static func reAdd(_ handler: @escaping (Any?) -> ()) -> ItemAction {
+        return ItemAction(
+            title: "Re-add",
```
Comment (dskuza) [inline PocketKit/Sources/PocketKit/ItemAction.swift:62]: Quick thought: maybe we shouldn't even use "Archive" / "Re-add" as titles, but "Move to My List" and "Move to Archive" and be more specific. Definitely a product question. It adds length, but also adds clarity imo.

File: PocketKit/Tests/PocketKitTests/ArchivedItemsViewModelTests.swift [hunk 3]
```diff
@@ -79,7 +79,11 @@ class ArchivedItemsViewModelTests: XCTestCase {
     }
 
     func test_deleteAction_delegatesToSource_andUpdatesSnapshot() {
-        source.stubDelete { _ in }
+        let expectDeleteCall = expectation(description: "expect a source.delete(archivedItem:)")
+        source.stubDelete { _ in
+            expectDeleteCall.fulfill()
+        }
+
```
Comment (dskuza) [inline PocketKit/Tests/PocketKitTests/ArchivedItemsViewModelTests.swift:86]: Love it. Back to the pre-predicate speeds we go!

**Review by dskuza — APPROVED** [review]
Great! Looks like our Source and Archive service setups have come in handy once again; this PR is super readable. 

➕ to finding a fix for the "flaky" predicate-based assertions.

I've left two non-blocking comments, just for opening up small convos.

File: PocketKit/Sources/PocketKit/ItemAction.swift (hunk 2 above)
Comment (jacobsimeon) [reply PocketKit/Sources/PocketKit/ItemAction.swift:62]: Yeah, I agree that the copy here could use some more thought. I went with "Re-add" per [recommendation](https://range.co/getpocket/checkins/AB8BFtGRHvac2_ZNDAAy?utm_medium=integration&utm_source=slack) from @marcin-kozinski but we can discuss with the team and tweak if necessary.

File: PocketKit/Tests/PocketKitTests/ArchivedItemsViewModelTests.swift (hunk 3 above)
Comment (jacobsimeon) [reply PocketKit/Tests/PocketKitTests/ArchivedItemsViewModelTests.swift:86]: 😎

File: PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift (hunk 1 above)
Comment (jacobsimeon) [reply PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift]: Good call, flipped 'em

**Pull request merged**